### PR TITLE
using login.portal.url to redirect users when password change

### PR DESCRIPTION
**Pull request by ArtificialOwl**
This will just add an entry in Personal with a link to the Portal.
This does not remove the current part of the Personal where you can change your password.

### CODE REVIEW

**Comment by daita** [comment]
selection_004

File: settings/templates/personal.php [hunk 1]
```diff
@@ -274,6 +274,19 @@
...
 ?>
 
+
+<?php
+if ($_['loginPortalUrl']) {
+?>
+	<form id="passwordportal" class="section" action="<?php p($_['loginPortalUrl'] . 'password-change'); ?>" method="GET">
+	<h2 class="inlineblock"><?php p($l->t('Password'));?></h2>
+		<br>
+		<input id="passwordbutton" type="submit" value="<?php p($l->t('Change your password from the Portal')); ?>" />
```
Comment (MorrisJobke) [inline settings/templates/personal.php]: I guess we should not use the term portal, because a user doesn't know this concept.

Comment (daita) [inline settings/templates/personal.php]: Yeah, also my english is random. Can we find a better text than `Change your password from the Portal` ?

**Comment by daita** [comment]
Also, I have no idea how to remove the current paragraph that allow to change password:

https://github.com/nextcloud/server/blob/master/lib/private/User/User.php#L321

File: settings/templates/personal.php (hunk 1 above)
Comment (MorrisJobke) [inline settings/templates/personal.php]: "Change password"

**Comment by MorrisJobke** [comment]
The sign-off was missing (see https://github.com/nextcloud/server/blob/master/CONTRIBUTING.md#sign-your-work) and I fixed it.

**Review by MorrisJobke — APPROVED** [review]

**Comment by nickvergessen** [comment]
Can we reuse https://github.com/nextcloud/server/blob/master/config/config.sample.php#L258 for this?

**Comment by MorrisJobke** [comment]
> Can we reuse https://github.com/nextcloud/server/blob/master/config/config.sample.php#L258 for this?

Would make sense. 👍

**Comment by daita** [comment]
What is this doing ? replacing the change password thing, or allowing a reset of your password from the login page of nextcloud ?

**Comment by nickvergessen** [comment]
It replaces the reset password form with a link to another website where this can be done, in cases where nc can not change the password. So sounds like it is the same

**Comment by MorrisJobke** [comment]
> It replaces the reset password form with a link to another website where this can be done, in cases where nc can not change the password. So sounds like it is the same

Ah no - this changes the password reset page, but not the password change. So it's not really the same.

**Comment by MorrisJobke** [comment]
@nickvergessen @LukasReschke Re-review please.

**Comment by MorrisJobke** [comment]
> This will just add an entry in Personal with a link to the Portal.
> This does not remove the current part of the Personal where you can change your password.

I would combine that. Either show the one or the other.

File: settings/templates/personal.php [hunk 2]
```diff
@@ -274,6 +274,19 @@
...
 ?>
 
+
+<?php
+if ($_['loginPortalUrl']) {
+?>
+	<form id="passwordportal" class="section" action="<?php p($_['loginPortalUrl'] . 'password-change'); ?>" method="GET">
+	<h2 class="inlineblock"><?php p($l->t('Password'));?></h2>
+		<br>
+		<input id="passwordbutton" type="submit" value="<?php p($l->t('Change your password')); ?>" />
```
Comment (nickvergessen) [inline settings/templates/personal.php]: use `<a class="button">` and kill the form please...

**Review by nickvergessen — CHANGES_REQUESTED** [review]

**Comment by LukasReschke** [comment]
I don't really like this approach, I'd either prefer this to be handled via an app or just as a patch sent to the customer in question…

**Review by LukasReschke — CHANGES_REQUESTED** [review]
I have been looking a little bit more into this and I believe we should solve it the following way:

1. Register a personal page with `\OCP\App::registerPersonal`
2. Use CSS or the JSLoader to move this up

```js
var settingsUrl = OC.generateUrl('/settings/personal');
if(window.location.pathname === settingsUrl) {
	var layout = $('#mySettingsDivChangeThis').detach();
	layout.insertAfter('#passwordform');
}
```

**Comment by MorrisJobke** [comment]
Let me look into this.

**Pull request closed without merging**